pkg/ruler/ring.go

```diff
@@ -0,0 +1,22 @@
+package ruler
+
+import (
+	"github.com/cortexproject/cortex/pkg/ring"
+)
+
+// CreateRulerRingConfig modifies a given ring config and repurposes it for use within
+// the ruler
+func CreateRulerRingConfig(cfg ring.Config) ring.Config {
+	// Copy the consul config struct and ensure a different prefix is used
+	consulCfg := cfg.Consul
+	consulCfg.Prefix = "rulers/"
+
+	return ring.Config{
+		Name:              "ruler", // Ensures separate ring metrics for ruler ring
+		Consul:            consulCfg,
+		Store:             cfg.Store,
+		HeartbeatTimeout:  cfg.HeartbeatTimeout,
+		ReplicationFactor: 1, // Ruler will never use a RF > 1
+		Mock:              cfg.Mock,
+	}
```

Comment:
Lets have the ring's registerflags func take a prefix so we can have multiple of them, and then lets use an independant one for the ruler.  What we've got here won't give us the operation flexibility.